minimizer/minimize.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+FILE="$1";
+CONDITION_SCRIPT="$2";
+shift;
+shift;
+CONDITION="$*";
```

Comment:
I'm not a giant fan of semicolons at the end of lines in bash, could you remove them in the rest of the program as well? If you keep it this way then either its more work the next time you modify it or it will be inconsistent if someone else doesn't add semicolons.
```suggestion
CONDITION_SCRIPT="$2"
shift
shift
CONDITION="$*"
```

Reply:
I'm adding them to stay close to other languages and as a safety guard that no accidental line-fusing can happen. Will do it though if its "idiomatic".

Reply:
> no accidental line-fusing can happen

Newlines in bash always count as a statement terminator (unlike JavaScript or Haskell, which have automatic semicolon insertion).

> Will do it though if its "idiomatic"

Yes, it looks pretty odd to me this way.